sig-platform-dev/README.md

```diff
@@ -0,0 +1,83 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+-->
+# Platform Dev Special Interest Group
+
+SIG Platform Dev (SIG-PD) is a cross-cutting SIG whose mission is to design, build, and maintain a cohesive, first-party Platform Development Kit (PDK) for developers who want to extend the Kubernetes platform by adding new APIs or enhancing existing APIs, as well as to advocate on behalf of such developers by driving feedback into other SIGs.
+
+## Meetings
+* Regular SIG Meeting: [Tuesdays at 10:00 PT (Pacific Time)]() (biweekly). [Convert to your timezone](http://www.thetimezoneconverter.com/?t=10:00&tz=PT%20%28Pacific%20Time%29).
+
+## Leadership
+
+### Chairs
+The Chairs of the SIG run operations and processes governing the SIG.
+
+* Jimmy Zelinskie (**[@jzelinskie](https://github.com/jzelinskie)**), Red Hat
+* Anthony Yeh (**[@enisoc](https://github.com/enisoc)**), Google
+
+### Technical Leads
+The Technical Leads of the SIG establish new subprojects, decommission existing
+subprojects, and resolve cross-subproject technical issues and decisions.
+
+* Jimmy Zelinskie (**[@jzelinskie](https://github.com/jzelinskie)**), Red Hat
+* Anthony Yeh (**[@enisoc](https://github.com/enisoc)**), Google
+* Phillip Wittrock (**[@pwittrock](https://github.com/pwittrock)**), Google
+
+## Contact
+* [Slack](https://kubernetes.slack.com/messages/sig-platform-dev)
+* [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-platform-dev)
+* [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/sig%2Fplatform-dev)
+
+## Subprojects
+
+The following subprojects are owned by sig-platform-dev:
+- **pdk-roadmap**
+  - Owners:
+    - https://raw.githubusercontent.com/kubernetes-sigs/pdk-roadmap/master/OWNERS
+- **sample-apiserver**
```

Comment:
What is our intention of owning these as subprojects.  The goal of the SIG is to build a toolkit for API extensions.  Would the samples be rewritten on top of this toolkit?

Reply:
I see the existing samples as a proto-PDK -- an early attempt to start a PDK within SIG-AM that didn't get very far yet because SIG-AM has many other components of its mission that took priority. By contrast, producing samples for use by non-core devs is a primary part of SIG-PD's mission, so I think we should own what exists now and push it towards a cohesive vision.

Reply:
👍 whilst I think demonstrating a "minimal" example is useful, ultimately best practice is the end-goal (and always has been).

That said, I don't think a simple switch to apiserver-builder right now is sufficient. Unless I misunderstand, apiserver-builder will not allow deep customisation of behaviour of a generated controller. This should be solved within the apiserver-builder repo before promotion to sample-*.

The general trend of moving components into apiextensions only emphasises this point imo. Perhaps dog-fooding is our bench mark - if we can write apiextensions-apiserver with apiserver-builder, maybe we are ready to recommend it as a best approach and switch more examples to it 😄 

That said, I am generally in favour of moving sample-* under the sig-pd charter in order to meet these goals.

Reply:
Next to being an example for 3rdparties to copy, sample-apiserver is also a testbed of using `k8s.io/apiserver`, i.e. it is a proof that the library can be used externally. Porting it to apiserver-builder neglects this 2nd purpose.

Reply:
@munnerz Thanks for the reply.  I wasn't suggesting apiserver-builder, but was more asking about what our goals with the samples are.  Do we want to keep these around as examples of how to use the apimachinery libraries directly?  If so, is SIG PD the correct owner or are the owners of the libraries themselves the correct owners?  If the goal of the SIG is to develop a toolkit is it ok to delete the samples and replace them with documentation on how to use the toolkit once it exists?

Reply:
> is it ok to delete the samples and replace them with documentation

No, compare my comment. Potentially we can move or rename them some day.

> replace them with documentation on how to use the toolkit once it exists?

It's early and not of much use to discuss those details before we have something like a toolkit or at least knowing how it will look like. I am not sold on the idea that we need a heavy layer on-top of `k8s.io/apiserver`. IMO we should make `k8s.io/apiserver` easy to use for the common use-case.

> If so, is SIG PD the correct owner

Enabling `k8s.io/{apiserver+code-generator}` for easy consumption is in scope of SIG PD for sure. SIG PD should own (and push to improve, **actively with PRs**, not only begging SIG AM for changes) the provided 3rdparty user-experience of the foundational libraries. Hence, owning the samples makes a lot of sense.

Reply:
@sttts 

> Next to being an example for 3rdparties to copy, sample-apiserver is also a testbed of using k8s.io/apiserver, i.e. it is a proof that the library can be used externally. Porting it to apiserver-builder neglects this 2nd purpose.

This seems like an argument for these to be owned by api machinery SIG.  SIGs should own their own tests.

> Enabling k8s.io/{apiserver+code-generator} for easy consumption is in scope of SIG PD for sure.

There are a number of ways to enable easy consumption.

> SIG PD should own (and push to improve, actively with PRs, not only begging SIG AM for changes) the provided 3rdparty user-experience of the foundational libraries.

I'd loved to see the upstream code-generators UX polished, and even submitted a proposal for doing so: https://github.com/kubernetes/community/pull/1315.  That said, because the stability of the code generators is important, it may be simpler and faster to develop UX that wraps the upstream libraries - e.g. what kubebuilder and apiserver-builder do.